app/mailers/kudo_mailer.rb

```diff
@@ -10,43 +10,24 @@ class KudoMailer < ActionMailer::Base
   #   guest_count: number of guest kudos
   def batch_kudo_notification(user_id, user_kudos)
     @commentables = []
-    @kudo_givers = {}
+    @user_kudos = JSON.parse(user_kudos)
     user = User.find(user_id)
     kudos_hash = JSON.parse(user_kudos)
 
     I18n.with_locale(Locale.find(user.preference.preferred_locale).iso) do
       kudos_hash.each_pair do |commentable_info, kudo_givers_hash|
-        # Parse the key to extract the type and id of the commentable - skip if no commentable
-        commentable_type, commentable_id = commentable_info.split('_')
+        # Parse the key to extract the type and id of the commentable so we can
+        # weed out any commentables that no longer exist.
+        commentable_type, commentable_id = commentable_info.split("_")
         commentable = commentable_type.constantize.find_by(id: commentable_id)
         next unless commentable
 
-        # If we have a commentable, extract names and process guest kudos text - skip if no kudos givers
-        names = kudo_givers_hash['names']
-        guest_count = kudo_givers_hash['guest_count']
-        kudo_givers = []
-
-        if !names.nil? && names.size > 0
-          kudo_givers = names
-          kudo_givers << guest_kudos(guest_count) unless guest_count == 0
-          @kudos_count = names.size + guest_count
-        else
-          kudo_givers << guest_kudos(guest_count).capitalize unless guest_count == 0
-          @kudos_count = guest_count
-        end
-        next if kudo_givers.empty?
-
         @commentables << commentable
-        @kudo_givers[commentable_info] = kudo_givers
       end
       mail(
         to: user.email,
-        subject: t('mailer.kudos.you_have', app_name: ArchiveConfig.APP_SHORT_NAME)
+        subject: t("mailer.kudos.you_have", app_name: ArchiveConfig.APP_SHORT_NAME)
```

Comment:
Lowkey want to fix the key names for this...

Reply:
Please do!